### PR TITLE
[VULKAN] Optimize Pipeline Lookups

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Reduces the number of times we have to look up a pipeline by only re-fetching pipelines if some pipeline-related state has changed.

### CODE REVIEW

File: src/FNA3D_Driver_Vulkan.c [hunk 1]
```diff
@@ -3934,17 +3938,29 @@ static VkPipeline VULKAN_INTERNAL_FetchPipeline(VulkanRenderer *renderer)
 
 static void VULKAN_INTERNAL_BindPipeline(VulkanRenderer *renderer)
 {
-	VkPipeline pipeline = VULKAN_INTERNAL_FetchPipeline(renderer);
+	VkShaderModule vertShader, fragShader;
+	MOJOSHADER_vkGetShaderModules(&vertShader, &fragShader);
 
-	if (pipeline != renderer->currentPipeline)
+	if (renderer->needNewPipeline ||
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_Vulkan.c:3944]: Tabs for ( and )

File: src/FNA3D_Driver_Vulkan.c [hunk 2]
```diff
@@ -6499,29 +6519,50 @@ static void VULKAN_SetBlendState(
...
 		driverData,
 		&blendState->blendFactor
 	);
+
 	VULKAN_SetMultiSampleMask(
 		driverData,
 		blendState->multiSampleMask
 	);
+
+	SDL_memcpy(&renderer->blendState, blendState, sizeof(FNA3D_BlendState));
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_Vulkan.c:6532]: Do we need to do an SDL_memcmp and request a new pipeline for changes here?

File: src/FNA3D_Driver_Vulkan.c [hunk 3]
```diff
@@ -6499,29 +6519,50 @@ static void VULKAN_SetBlendState(
...
 	FNA3D_DepthStencilState *depthStencilState
 ) {
 	VulkanRenderer *renderer = (VulkanRenderer*) driverData;
-	SDL_memcpy(
-		&renderer->depthStencilState,
-		depthStencilState,
-		sizeof(FNA3D_DepthStencilState)
-	);
+
+	if (renderer->depthStencilState.stencilMask != depthStencilState->stencilMask ||
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_Vulkan.c:6541]: Since we're storing with memcpy we can probably replace all this with memcmp to keep it short.

Comment (thatcosmonaut) [reply src/FNA3D_Driver_Vulkan.c:6541]: Oh, i didn't know about memcmp. Neat! I can rewrite some of these to use that func instead

File: src/FNA3D_Driver_Vulkan.c (hunk 2 above)
Comment (thatcosmonaut) [reply src/FNA3D_Driver_Vulkan.c:6532]: the comparisons and pipeline checks happen in the above functions, not sure if memcmp is worth doing before the copy?

Comment (thatcosmonaut) [reply src/FNA3D_Driver_Vulkan.c:6532]: Oh wait, I see, yes, this is an error actually

**Pull request merged**